ui/js/directives/perf/compare.js

```diff
@@ -20,26 +19,43 @@ treeherder.directive(
                 function filter(item, matchText) {
                     return !matchText || item.toLowerCase().indexOf(matchText.toLowerCase()) > (-1);
                 }
-                scope.filterTest = function(item) {
-                    return filter(item, scope.testFilter);
-                };
-                scope.filterPlatform = function(result) {
-                    return filter(result.name, scope.platformFilter) &&
-                        (!scope.showOnlyImportant || result.isMeaningful) &&
-                        (!scope.showOnlyConfident || result.isConfident) &&
-                        (scope.showUnreliablePlatforms || !_.contains(
+                function hidenOption(result) {
+                    return (!scope.showOnlyImportant || result.isMeaningful)
+                        && (!scope.showOnlyConfident || result.isConfident)
+                        && (scope.showUnreliablePlatforms || !_.contains(
                             phUnreliablePlatforms, result.name));
+                }
+                scope.filterResult = function(results, key) {
```

Comment:
This doesn't need to be assigned to the scope variable, as it's only used inside the direct afaict. 

So just say function ```filterResult(results, key) { ...```